File: js_project/Strings/AllSubstringsGenerator.py
s1 = "abc"
length = len(s1)
print ("Generating all possible substrings for \"" + s1 + "\":\n")
for start in range (0, length):
    for end in range (start , length):
        sub = ""
        for i in range (start , end+1 ):
            sub = sub + s1[i]
        print ("Start: " + str(start) + ", End: " + str(end) + " -> \"" + sub + "\"")

File: js_project/Strings/AllSubstringsGenerator.js
let s1 = "abc";
let length = s1.length;

console.log("Generating all possible substrings for \"" + s1 + "\":\n");

for (let start = 0; start < length; start++) {
    for (let end = start; end < length; end++) {
        let sub = "";
        
        for (let i = start; i <= end; i++) {
            sub = sub + s1.charAt(i);
        }
        console.log("Start: " + start + ", End: " + end + " -> \"" + sub + "\"");
    }
}